pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -60,21 +63,22 @@ func (Plugin) OnInboundListener(in *plugin.InputParams, mutable *networking.Muta
 		// Only care about sidecar.
 		return nil
 	}
-	return buildFilter(in, mutable)
+	return buildFilter(in, mutable, false)
 }
 
-func buildFilter(in *plugin.InputParams, mutable *networking.MutableObjects) error {
+func buildFilter(in *plugin.InputParams, mutable *networking.MutableObjects, isPassthrough bool) error {
 	ns := in.Node.Metadata.Namespace
 	applier := factory.NewPolicyApplier(in.Push, ns, labels.Collection{in.Node.Metadata.Labels})
-	if mutable.Listener == nil || (len(mutable.Listener.FilterChains) != len(mutable.FilterChains)) {
-		return fmt.Errorf("expected same number of filter chains in listener (%d) and mutable (%d)", len(mutable.Listener.FilterChains), len(mutable.FilterChains))
-	}
 	endpointPort := uint32(0)
 	if in.ServiceInstance != nil {
 		endpointPort = in.ServiceInstance.Endpoint.EndpointPort
 	}
 
-	for i := range mutable.Listener.FilterChains {
+	for i := range mutable.FilterChains {
+		if isPassthrough {
+			// Get the real port from the filter chain match if this is generated for pass through filter chain.
```

Comment:
Also wondering if this new meaning (to me) of "pass through" is consistent across code base.

Reply:
> If so, I feel we are overloading the Passthrough naming a bit, which used for all traffic before, now it's also for specific ports, right? I see reasons because the plugin interfaces are this way, OnPassThroughFilterChains.

Yes,  all other settings will be the same as the pass through filter chain except the port part. This happens only when the port is not defined in service but specified in PeerAuthN (per-port level setting), previously we just ignore it.

And thinking of this, should we add a release note for this? It's unfortunate that we didn't document it clearly that the per-port PeerAuthN is not supported on pass through filter chain before.

Reply:
yeah, definitely a release note. Important change, IMO.

Reply:
added release note.